hack/embed-pinniped-binary.sh

```diff
@@ -14,8 +14,23 @@ set -o pipefail
 set -o xtrace
 
 GO=$1
-GOOS=$2
-GOARCH=$3
+shift
+GOOS=$1
+shift
+GOARCH=$1
+shift
 
-cd pinniped && GOARCH=${GOARCH} GOOS=${GOOS} ${GO} build -o ../cmd/cli/plugin/pinniped-auth/asset/pinniped ./cmd/pinniped && cd ..
-git update-index --assume-unchanged cmd/cli/plugin/pinniped-auth/asset/pinniped
+while (( "$#" )); do
```

Comment:
Did we want to consider a top-level "tmp" subdirectory, rather than in the top-level directory of the repo, just because it's a little confusing seeing 'pinniped' cloned here at a later time when you don't expect it?